### PR TITLE
Simplify and improve Libvirt event notification

### PR DESCRIPTION
**Pull request by rmohr**
Another attempt to fix THE flaky CI test.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

File: pkg/virt-handler/virtwrap/cache/cache.go [hunk 1]
```diff
@@ -187,71 +187,47 @@ func NewSharedInformer(c virtwrap.Connection) (cache.SharedInformer, error) {
...
-			domain.SetState(api.NoState, api.ReasonUnknown)
-		} else {
-			domain.SetState(convState(status), convReason(status, reason))
-		}
-	default:
-		spec, err := NewDomainSpec(d)
-		if err != nil {
-			logging.DefaultLogger().Error().Reason(err).Msg("Could not fetch the Domain specification.")
+		if !virtwrap.IsNotFound(err) {
+			logging.DefaultLogger().Error().Object(domain).Reason(err).Msg("Could not fetch the Domain specification.")
```
Comment (mpolednik) [inline pkg/virt-handler/virtwrap/cache/cache.go]: When you're at it, this could be moved to some small helper function since the same code is used below.

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by fabiand** [comment]
retest this please

**Comment by stu-gott** [comment]
What's the status of this at this point?

**Comment by davidvossel** [comment]
@rmohr I see a couple of optimizations here, but nothing that should technically fix anything.  Can we close this, or did you wan to follow up with this further?

**Comment by rmohr** [comment]
Rebased it. @davidvossel I would feel better, if we bring it in, since it is more robust and easier, although it does not explicitly fix any test.

**Comment by rmohr** [comment]
retest this please

**Comment by davidvossel** [comment]
> Rebased it. @davidvossel I would feel better, if we bring it in, since it is more robust and easier, although it does not explicitly fix any test.

agreed.

**Pull request merged**